pallets/registrar/src/benchmark_blob.rs

```diff
@@ -0,0 +1,162 @@
+    206, 62, 80, 169, 1, 159, 12, 182, 243, 110, 2, 122, 187, 32, 145, 207, 181, 17, 10, 181, 8,
+    127, 137, 0, 104, 95, 6, 21, 91, 60, 217, 168, 201, 229, 233, 162, 63, 213, 220, 19, 165, 237,
+    32, 0, 0, 0, 0, 0, 0, 0, 0, 104, 95, 8, 49, 108, 191, 143, 160, 218, 130, 42, 32, 172, 28, 85,
+    191, 27, 227, 32, 0, 0, 0, 0, 0, 0, 0, 0, 128, 254, 108, 203, 37, 75, 132, 240, 210, 32, 46,
+    181, 5, 189, 223, 159, 84, 203, 158, 189, 15, 178, 113, 144, 114, 233, 46, 229, 124, 29, 161,
+    216, 9,
+];
+
+pub fn benchmark_blob() -> BenchmarkBlob {
+    #[allow(const_item_mutation)]
+    BenchmarkBlob::decode(&mut ENCODED_BLOB).unwrap()
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        crate::{
+            mock::{get_ed25519_pairs, new_test_ext, run_to_block, ParaRegistrar, ALICE},
+            ParaInfo, REGISTRAR_PARAS_INDEX,
+        },
+        cumulus_test_relay_sproof_builder::RelayStateSproofBuilder,
+        frame_support::Hashable,
+        parity_scale_codec::Encode,
+        sp_core::{crypto::AccountId32, Decode, Pair},
+        tp_traits::ParaId,
+    };
+
+    #[test]
+    #[ignore = "used to generate benchmark data"]
+    fn benchmark_blob_generate_data() {
+        new_test_ext().execute_with(|| {
+            run_to_block(1);
+
+            let pairs = get_ed25519_pairs(1);
+            let mut sproof = RelayStateSproofBuilder::default();
+            let para_id: ParaId = 0.into();
+            let bytes = para_id.twox_64_concat();
+            let key = [REGISTRAR_PARAS_INDEX, bytes.as_slice()].concat();
+            let para_info: ParaInfo<
+                cumulus_primitives_core::relay_chain::AccountId,
+                cumulus_primitives_core::relay_chain::Balance,
+            > = ParaInfo {
+                manager: pairs[0].public().into(),
+                deposit: Default::default(),
+                locked: None,
+            };
+            sproof.additional_key_values = vec![(key, para_info.encode())];
+            let (relay_parent_storage_root, proof) = sproof.into_state_root_and_proof();
+
+            let account = ALICE;
+            let relay_storage_root = relay_parent_storage_root;
+            let signature_msg =
+                ParaRegistrar::relay_signature_msg(para_id, &account, relay_storage_root);
+            let signature_account_u64: cumulus_primitives_core::relay_chain::Signature =
+                pairs[0].sign(&signature_msg).into();
+
+            // Account generated by
+            // create_funded_user::<T>("caller", 0, T::DepositAmount::get());
+            let account32 = AccountId32::decode(
+                &mut [
+                    92u8, 119, 3, 63, 206, 138, 144, 69, 130, 74, 102, 144, 187, 249, 156, 109,
+                    178, 105, 80, 47, 10, 141, 29, 42, 0, 133, 66, 213, 105, 10, 7, 73,
+                ]
+                .as_slice(),
+            )
+            .unwrap();
```

Comment:
An idea to make these run faster:remove the y parameteres in both register and deregister. In the `register` benchmarks we can completely ignore the y parameter as we have a mapping now for the chains pending verification.

In the `deregister` extrinsic, let's assume x and y is always max, since it's what we are doing anyways in the pallet. That way we iterate only over z